modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java

```diff
@@ -2032,10 +2049,28 @@ protected String getJavaOptsString() {
     }
 
     /**
-     * @return the value of an environment variable. Overriden for testing.
+     * @return the value of an environment variable
      * @since 9.1
+     * @apiNote exists to be overridden by tests
      */
-    protected String getEnvironmentVariableValue(String key) {
+    protected String getEnvironment(String key) {
         return System.getenv(key);
     }
+
+    /**
+     * @return the value of an environment variable
+     * @since 11.1
+     * @see #getEnvironment(String)
+     */
+    protected String getEnvironment(String key, String defaultValue) {
+        return Objects.requireNonNullElse(getEnvironment(key), defaultValue);
+    }
+
+    /**
+     * @return the nuxeo.defaults file for current {@code NUXEO_ENVIRONMENT}
+     * @since 11.1
+     */
+    protected String getNuxeoEnvironmentConfName() {
+        return String.format(NUXEO_ENVIRONMENT_CONF_FORMAT, getEnvironment(NUXEO_ENVIRONMENT));
```

Comment:
wouldn't you want to use the other signature of getEnvironment() with a default ?
The signature with a default is used in the public setProperties() method (line 1677) so it may be more consistent ?
The documentation says this signature (without a default) is only for tests. But it seems to be used in several internal methods. Maybe it's outdated ?
> * @apiNote exists to be overridden by tests

Reply:
There're several things in your comment.

>wouldn't you want to use the other signature of getEnvironment() with a default ?

Which default do you want to use?

>The signature with a default is used in the public setProperties() method (line 1677) so it may be more consistent?

Because I use the `String#isBlank` method and so I can't have null.

>The documentation says this signature (without a default) is only for tests. But it seems to be used in several internal methods. Maybe it's outdated?
@APinote exists to be overridden by tests

The sentence doesn't say it's used by tests, it says that the method exists in order to let let test override it to provide a mock (see the tests).

Reply:
1- I assumed this is to avoid null so I would have given `""` like line 1677 (like you explain in your second point
2- yes
3 - Ok thanks for the explenation !

Reply:
Probably, but this is not needed here. Furthermore, if the code evolves, I prefer to have `nuxeo.null` which clearly shows there's something wrong than `nuxeo.`.